### PR TITLE
[WIP] Break up crosslink_committees_at_slot

### PR DESCRIPTION
**Pull request by djrtwo**
Attempt to "simplify" `get_crosslink_committees_at_slot`.
Broke it up into (maybe) more manageable functions. Not really convinced this helps much but throwing it up here while I think more

cc: @JustinDrake, @terenc3t

### CODE REVIEW

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -848,6 +849,108 @@ def get_next_epoch_committee_count(state: BeaconState) -> int:
 
 ### `get_crosslink_committees_at_slot`
```
Comment (terenc3t) [inline specs/core/0_beacon-chain.md]:
```suggestion
### `get_crosslink_committees`
```

Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: done

File: specs/core/0_beacon-chain.md [hunk 2]
```diff
@@ -846,6 +847,108 @@ def get_next_epoch_committee_count(state: BeaconState) -> int:
...
+### `get_crosslink_committees`
+
+```python
+def get_crosslink_committees(state: BeaconState,
+                             seed: Bytes32,
+                             shuffling_epoch: Epoch,
+                             slot: Slot,
+                             start_shard: Shard,
+                             committees_per_epoch: int) -> List[Tuple[List[ValidatorIndex], Shard]]:
+    offset = slot % SLOTS_PER_EPOCH
```
Comment (terenc3t) [inline specs/core/0_beacon-chain.md]: should we add an assert to check `shuffling_epoch` out of bound condition?

`assert get_previous_epoch(state)  <= shuffling_epoch <= get_current_epoch(state)+1`

File: specs/core/0_beacon-chain.md [hunk 3]
```diff
@@ -846,6 +847,108 @@ def get_next_epoch_committee_count(state: BeaconState) -> int:
...
+### `get_next_epoch_committees_at_slot`
+
+```python
+def get_next_epoch_committees_at_slot(state: BeaconState,
+                                      slot: Slot,
+                                      registry_change: bool) -> List[Tuple[List[ValidatorIndex], Shard]]:
+    epochs_since_last_registry_update = current_epoch - state.validator_registry_update_epoch
+    if registry_change:
+        committees_per_epoch = get_next_epoch_committee_count(state)
+        seed = generate_seed(state, next_epoch)
```
Comment (terenc3t) [inline specs/core/0_beacon-chain.md]: we need to define `next_epoch`

File: specs/core/0_beacon-chain.md [hunk 4]
```diff
@@ -846,6 +847,108 @@ def get_next_epoch_committee_count(state: BeaconState) -> int:
...
+        seed = state.current_shuffling_seed
+        shuffling_epoch = state.current_shuffling_epoch
+        shuffling_start_shard = state.current_shuffling_start_shard
+
+    return get_crosslink_committees(
+        state,
+        seed,
+        shuffling_epoch,
+        slot,
+        start_shard,
```
Comment (terenc3t) [inline specs/core/0_beacon-chain.md]:
```suggestion
        shuffling_start_shard,
```

File: specs/core/0_beacon-chain.md [hunk 5]
```diff
@@ -866,47 +969,11 @@ def get_crosslink_committees_at_slot(state: BeaconState,
...
-    slot_start_shard = (shuffling_start_shard + committees_per_slot * offset) % SHARD_COUNT
-
-    return [
-        (
-            shuffling[committees_per_slot * offset + i],
-            (slot_start_shard + i) % SHARD_COUNT,
-        )
-        for i in range(committees_per_slot)
-    ]
+        return get_next_epoch_committee_count(state, slot, registry_change)
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]:
```suggestion
        return get_next_epoch_committees_at_slot(state, slot, registry_change)
```

**Comment by terenc3t** [comment]
Implemented here: https://github.com/prysmaticlabs/prysm/pull/1771

**Comment by JustinDrake** [comment]
Closing in favour of #808 🔥

**Pull request closed without merging**